cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/ProgramController.java

```diff
@@ -115,7 +121,9 @@ public ProgramRunStatus getRunStatus() {
 
   ListenableFuture<ProgramController> resume();
 
-  ListenableFuture<ProgramController> stop();
+  ListenableFuture<ProgramController> complete();
```

Comment:
Whenever something call `ProgramController.stop`. It will stop (kill) the running program. When the program that finished by itself, the controller of it will transit into `Completed` state. No one should ever need to call any method on `ProgramController`. How it is implemented internally is independent of what the interface contract.